core/scc/lscc/deployedcc_infoprovider.go

```diff
@@ -84,6 +85,34 @@ func (p *DeployedCCInfoProvider) ChaincodeInfo(channelName, chaincodeName string
 	}, nil
 }
 
+// AllChaincodesInfo returns a list of legacy chaincodes info
+func (p *DeployedCCInfoProvider) AllChaincodesInfo(channelName string, qe ledger.SimpleQueryExecutor) ([]*ledger.DeployedChaincodeInfo, error) {
+	iter, err := qe.GetStateRangeScanIterator(lsccNamespace, "", "")
+	if err != nil {
+		return nil, err
+	}
+	result := make([]*ledger.DeployedChaincodeInfo, 0)
+	for {
+		entry, err := iter.Next()
+		if err != nil {
+			return nil, err
+		}
+		if entry == nil {
+			break
+		}
+
+		kv := entry.(*queryresult.KV)
+		if !privdata.IsCollectionConfigKey(kv.Key) {
+			deployedccInfo, err := p.ChaincodeInfo(channelName, kv.Key, qe)
```

Comment:
In case you have not already figured, the entries for the chaincodes and collection config could be interleaved. So, I feel that better to either keep this function as is for simplicity or otherwise maintain a map that keeps one entry for each chaincode. This map would get updated by either adding a new entry for the chaincodeInfo, or by updating the existing chaincodeInfo with its collection config (when encountered). I do not have a strong inclination either way, as this is not in the critical perf path.

Reply:
What @manish-sethi said was the exact reason that the PR skipped collection entries and called ChaincodeInfo for a simple implementation. Otherwise, we will have to assemble multiple entries.

Reply:
Okay. It just looked weird to me that the required key and value are present in this function but we call some other function to get them again. As we have provided it like a generic API, I cannot comment on the performance as I don't know where this API would be called in the future. 

Anyway, I am fine with the current usage as it helps to avoid the following lines of code. Note that it is always possible to reduce the number of lines of code by adding hacks. 

```
deployedCCInfo := make(map[string]*DeployedChaincodeInfo)
if  !privdata.IsCollectionConfigKey(kv.Key) {
        chaincodeData := &ccprovider.ChaincodeData{}
	if err := proto.Unmarshal(chaincodeDataBytes, chaincodeData); err != nil {
		return nil, errors.Wrap(err, "error unmarshalling chaincode state data")
	}
	deployedCCInfo[kv.Key] = &ledger.DeployedChaincodeInfo{Name: kv.Key, Hash: chaincodeData.Id, Version: chaincodeData.Version, IsLegacy: true}
        continue
}
ccName := privdata.GetCCNameFromCollectionConfigKey(kv.Key)
collectionConfigPkg := &peer.CollectionConfigPackage{}
if err := proto.Unmarshal(collectionConfigPkgBytes, collectionConfigPkg); err != nil {
	return nil, errors.Wrap(err, "error unmarshalling chaincode collection config pkg")
}
deployedCCInfo[ccName].ExplicitCollectionConfigPkg = collectionConfigPkg

deployedCCInfo := []*DeployedChaincodeInfo{}
for _, ccInfo := range deplyedCCInfo {
        append(deployedCCInfo, ccInfo)
}
```